### ISSUE TITLE
Unmatched bm_cube.shape between hera_sim and vis_cpu (A bug?)

### ISSUE THREAD

**Issue by JianrongTan** [HERA-Team/hera_sim#177]
I went into some errors when trying to run polarized simulations with hera_sim.  
```
File "/users/jtan/hera_sim/hera_sim/visibilities/vis_cpu.py", line 386, in _base_simulate
    vis = self._vis_cpu(
  File "/lustre/aoc/projects/hera/jtan/anaconda3/envs/tests/lib/python3.9/site-packages/vis_cpu/vis_cpu.py", line 168, in vis_cpu
    assert bm_cube.shape == (nax, nfeed, nant, bm_pix, bm_pix), (
AssertionError: bm_cube must have shape (NAXES, NFEEDS, NANTS, BM_PIX, BM_PIX) if polarized=True. Shape wanted: (2, 2, 19, 100, 100); shape given: (19, 2, 2, 100, 100)
```

By checking the code of `hera_sim.visibilities.vis_cpu._base_simulate`, I find with `self.polarized == True`, `_beam_lm` is first initialized into a shape of (NANT, NAXES, NFEEDS, BM_PIX, BM_PIX) and then passed to `self._vis_cpu` (which is `vis_cpu.vis_cpu` in this case): 

```
vis = self._vis_cpu(
                 antpos=self.antpos,
                 freq=freq,
                 eq2tops=eq2tops,
                 crd_eq=crd_eq,
                 I_sky=I_sky[i],
                 beam_list=beam_list if not self.use_pixel_beams else None,
                 bm_cube=_beam_lm if self.use_pixel_beams else None,
                 precision=self._precision,
                 polarized=self.polarized,
             )
```

While in `vis_cpu.vis_cpu`, it has such a sentence:
```
if polarized:
            assert bm_cube.shape == (nax, nfeed, nant, bm_pix, bm_pix)
```
So `_beam_lm` generated in `_base_simulate` doesn't match the shape of `bm_cube` asked by `vis_cpu.vis_cpu`. I am not sure if I just caught a bug here.

**Comment by steven-murray** [comment]
Hi @JianrongTan, sorry that you ran into this. This is all being fixed by #175. I was hoping to have that PR merged by now, but it turns out polarization is tricky, and I have very limited time to work on it til end of next week.

**Comment by JianrongTan** [comment]
Thanks @steven-murray! 
Fixed in #175
